### PR TITLE
Support Jenkins credentials of type SSHUserPrivateKey in maven settings.xml

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
Support Jenkins credentials of type SSHUserPrivateKey in maven settings files in addition to the existing support of UsernamePassword credentials

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
I cannot reproduce the build failure on my computer. Likely to be a random failure.

**Comment by cyrille-leclerc** [comment]
I pushed a minor change in the javadocs and the build get green :-)

**Comment by cyrille-leclerc** [comment]
@imod hi, could you have a look at this?

File: src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java [hunk 1]
```diff
@@ -73,6 +78,7 @@ private CredentialsHelper() {
     }
 
     public static List<StandardUsernameCredentials> findValidCredentials(final String serverIdPattern) {
+        // TODO Cyrille Le Clerc 2016/05/08: why do we return an empty list of server id is empty?
```
Comment (imod) [inline src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java]: what else would you expect? returning `null` might just cause NPEs or null checks in other places

**Comment by imod** [comment]
@cyrille-leclerc sorry I did not see this PR earlier! This is a great addition and I think it looks good.
👍

File: src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java (hunk 1 above)
Comment (cyrille-leclerc) [inline src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java]: I should have clarified, why not returning all the StandardUsernameCredentials and not apply domain restriction?

**Comment by cyrille-leclerc** [comment]
@imod do you think that we can merge this PR? Or you want to clarify my "TODO"?

**Comment by cyrille-leclerc** [comment]
Hello @imod, could you review my last comment?

**Comment by imod** [comment]
to be honest, I did not have the time yet - I think I can take a look next week

File: src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java (hunk 1 above)
Comment (imod) [inline src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java]: I guess we could also just return all - true. Will you adjust the PR to reflect it?

**Comment by imod** [comment]
@cyrille-leclerc I think we can do it as you say: return all `StandardUsernameCredentials` will you adapt your PR to also include this?

**Comment by cyrille-leclerc** [comment]
@imod I have updated the code so that the credentials drop down list will return all the StandardUsernameCredentials if the maven server id is not yet defined.

**Comment by cyrille-leclerc** [comment]
Thanks @imod

**Comment by imod** [comment]
nop!
referencing the issue: https://issues.jenkins-ci.org/browse/JENKINS-33165

**Pull request merged**